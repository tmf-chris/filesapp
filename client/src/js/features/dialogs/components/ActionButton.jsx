import React from 'react';
import Button from '@material-ui/core/Button';
import CloudUploadIcon from '@material-ui/icons/CloudUpload';

const styles = {
    button: {
        marginLeft: '12px'
    }
};

const ActionButton = ({ action, label }) =>
    <div>
        <Button
            variant="contained"
            color="default"
            onClick={action}
        >
            { label }
            <CloudUploadIcon
                style={styles.button}
            />
        </Button>
    </div>;

export default ActionButton;