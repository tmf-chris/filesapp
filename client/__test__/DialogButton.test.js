import React from 'react';
import ActionButton from '../src/js/features/dialogs/components/ActionButton';
import renderer from 'react-test-renderer';

test('Action button', () => {
    const component = renderer.create(
        <ActionButton action={(e) => {}} label='Button'/>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});