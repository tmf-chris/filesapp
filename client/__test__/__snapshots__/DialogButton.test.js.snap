// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action button 1`] = `
<div>
  <button
    className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-raised-15"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-2"
    >
      Button
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-30"
        focusable="false"
        style={
          Object {
            "marginLeft": "12px",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM14 13v4h-4v-4H7l5-5 5 5h-3z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root-37"
    />
  </button>
</div>
`;
